trieste/models/keras/models.py

```diff
@@ -377,6 +378,62 @@ def update_encoded(self, dataset: Dataset) -> None:
         """
         return
 
+    def prepare_tf_data(
+        self,
+        x: dict[str, TensorType],
+        y: dict[str, TensorType],
+        batch_size: int,
+        num_points: int,
+        validation_split: float = 0.0,
+    ) -> Union[tf.data.Dataset, tuple[tf.data.Dataset, tf.data.Dataset]]:
```

Comment:
might be nicer to always return a tuple?

```suggestion
    ) -> tuple[tf.data.Dataset, Optional[tf.data.Dataset]]]:
```